### DISCUSSION TITLE
HW 2 Part 2.2

### DISCUSSION THREAD

**Discussion by Aarons-Joshua** [cooperrc/computational-mechanics#38] [Q&A]
Has anyone figured out how to separate just the values for the years 2017 in the data file? I am having a hard time figuring out how to filter out only the data that has a date of introduction in 2017 for usage in my analysis and box plots. Please let me know if anyone knows how to do this, it would be greatly appreciated!

**Comment by Aidan-Myatt** [comment]
This is how I did it personally. I am sure there is a better way, but this is what worked for me.

![image](https://user-images.githubusercontent.com/106316906/172652222-5c87aad2-1faa-4b20-a4d5-f389ea18f649.png)

**Reply by cooperrc** [reply]
Nice!

This is a great way to visually pick out the data(sometimes its easier especially if most of your data processing is Excel-work)

Try this instead

```python
trans[trans['Date of Introduction'] == 2017]
```

You'll a _new_ Pandas dataframe that satisfies the condition that the 'Date of Introduction' `== 2017`. I usually create a new variable, that I can do math on after e.g. 


```python
trans2017 = trans[trans['Date of Introduction'] == 2017]
trans2017['MOS transistor count'].mean()
```
